### PR TITLE
Add component catalog guide to VPE

### PR DESCRIPTION
**Pull request by karlaspuldaro**
Closes #2281 
Add guidance about how to add to the component palette from an empty pipeline.

### What changes were proposed in this pull request?
- [x] Override empty canvas with custom message guide
- [x] Add link to [example components repo ](https://github.com/elyra-ai/examples#custom-pipeline-component-examples)
- [x] Bring back drag and drop icon and styling used in 2.2.4 release
- [x] Open component catalog widget from text icon
- [x] Create separate react component

New message (Generic vs runtime specific empty pipeline)
image

### How was this pull request tested?

Manual testing
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: packages/pipeline-editor/src/EmptyPipelineContent.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import { componentCatalogIcon, dragDropIcon } from '@elyra/ui-components';
+import { ILabShell } from '@jupyterlab/application';
+
+import React from 'react';
+
+import { COMPONENT_CATALOGS_SCHEMASPACE } from './PipelineService';
+
+const HEADER_CLASS = 'empty-pipeline-header';
+
+const COMPONENT_EXAMPLES_URLl =
```
Comment (bourdakos1) [inline packages/pipeline-editor/src/EmptyPipelineContent.tsx:26]: is `URLl` a typo?

**Review by bourdakos1 — APPROVED** [review]
Code LGTM

File: packages/pipeline-editor/src/EmptyPipelineContent.tsx (hunk 1 above)
Comment (karlaspuldaro) [reply packages/pipeline-editor/src/EmptyPipelineContent.tsx:26]: It is! lol thx

File: packages/pipeline-editor/src/PipelineEditorWidget.tsx [hunk 2]
```diff
@@ -1102,7 +1102,13 @@ const PipelineWrapper: React.FC<IProps> = ({
...
           leftPalette={true}
         >
-          <EmptyPipelineContent shell={shell} />
+          <EmptyPipelineContent
+            openComponentCatalog={(): void => {
+              shell.activateById(
+                `elyra-metadata:${COMPONENT_CATALOGS_SCHEMASPACE}`
+              );
+            }}
+          />
```
Comment (bourdakos1) [inline packages/pipeline-editor/src/PipelineEditorWidget.tsx:1111]: Nice! I like this better :)

**Comment by ptitzler** [comment]
Custom components are not applicable to generic pipelines. Is there a way to only show the extended version of the text in the KFP/AA editors?

File: packages/pipeline-editor/src/EmptyPipelineContent.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      <dragDropIcon.react
+        className="drag-drop-icon"
+        tag="div"
+        elementPosition="center"
+        height="120px"
+      />
+      <div className={HEADER_CLASS}>
+        <h3>
+          Start your new pipeline by dragging files from the file browser pane
+          or by adding a custom component clicking the{' '}
```
Comment (ptitzler) [inline packages/pipeline-editor/src/EmptyPipelineContent.tsx:44]: We should use plural for consistency

```suggestion
          or by adding custom components clicking the{' '}
```

**Comment by karlaspuldaro** [comment]
Thx @ptitzler, I addressed your comments and updated the screenshot in the PR description.

File: packages/pipeline-editor/src/PipelineEditorWidget.tsx [hunk 4]
```diff
@@ -1100,7 +1102,19 @@ const PipelineWrapper: React.FC<IProps> = ({
...
+            openComponentCatalog={
+              pipelineRuntimeType !== 'Generic'
+                ? (): void => {
+                    shell.activateById(
+                      `elyra-metadata:${COMPONENT_CATALOGS_SCHEMASPACE}`
+                    );
+                  }
+                : undefined
+            }
+          />
```
Comment (bourdakos1) [inline packages/pipeline-editor/src/PipelineEditorWidget.tsx:1116]: Can we break this into two separate components?

Something like:
```tsx
const handleOpenCatalog = () => {
  shell.activateById(`elyra-metadata:${COMPONENT_CATALOGS_SCHEMASPACE}`)
}

{type === undefined ? (
  <EmptyGenericPipeline />
) : (
  <EmptyPlatformSpecificPipeline onOpenCatalog={handleOpenCatalog} />
)}
```

> **Note:** I think you need to rebase, the runtime type stuff has all changed. In this code snippet I use `type` which is `undefined` for "Generic"

**Review by bourdakos1 — CHANGES_REQUESTED** [review]

**Review by bourdakos1 — APPROVED** [review]
lgtm

**Review by ptitzler — APPROVED** [review]
LGTM

**Pull request merged**